### PR TITLE
Update extract_score_and_reasons to work across providers

### PR DESCRIPTION
**Pull request by joshreini1**
Update structured extraction and COT prompting for better results across providers e.g. Bedrock, Ollama

Fixes: https://github.com/truera/trulens/issues/716 and open email with amazon team, possibly: https://github.com/truera/trulens/issues/609

### CODE REVIEW

**Comment by piotrm0** [comment]
I think the more structured messages should be preferable if available so I'm not sure about just using prompts always.

**Comment by joshreini1** [comment]
> I think the more structured messages should be preferable if available so I'm not sure about just using prompts always.

Good point - updated to use prompt only for bedrock.

File: trulens_eval/trulens_eval/feedback/provider/bedrock.py [hunk 1]
```diff
@@ -63,17 +61,64 @@ def _create_chat_completion(
         assert self.endpoint is not None
         assert prompt is not None, "Bedrock can only operate on `prompt`, not `messages`."
 
-        # NOTE(joshr): only tested with sso auth
         import json
 
-        body = json.dumps({"inputText": prompt})
+        body = json.dumps({"inputText": prompt, "textGenerationConfig": {
+                "maxTokenCount": 4096,
```
Comment (piotrm0) [inline trulens_eval/trulens_eval/feedback/provider/bedrock.py:67]: Make a task to make these parameters configurable in a future PR?

**Review by piotrm0 — APPROVED** [review]

File: trulens_eval/trulens_eval/feedback/provider/bedrock.py (hunk 1 above)
Comment (piotrm0) [reply trulens_eval/trulens_eval/feedback/provider/bedrock.py:67]: @joshreini1 approved but see this note.

Comment (joshreini1) [reply trulens_eval/trulens_eval/feedback/provider/bedrock.py:67]: Added a TODO for this

**Pull request merged**